### PR TITLE
feat(Core/DBC): Implement NamesProfanity and NamesReserved DBC

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Implement NamesReserved.dbc
-  Implement NamesProfanity.dbc
- Move DB reserved_names check into CheckPlayerName()

## Issues Addressed:

- Closes 

## Tests Performed:

- Tested in-game
- 

## How to Test the Changes:

1. Try to create a character with a name that is within one of these DBCs

i.e 
`bollocks` - NamesProfanity
`Thrall` - NamesReserved

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/shared/DataStores/DBCStructure.h [hunk 1]
```diff
@@ -1377,6 +1377,20 @@ struct MovieEntry
     //uint32      unk2;                                     // 2 always 100
 };
 
+struct NamesReservedEntry
+{
+    //uint32         ID;                                    // 0
+    char const*      Pattern;                               // 1
+    //uint32         Languague;                             // 2
```
Comment (heyitsbench) [inline src/server/shared/DataStores/DBCStructure.h:1384]: Language.

Comment (Kitzunu) [reply src/server/shared/DataStores/DBCStructure.h:1384]: Kek, sometimes my fingers are fast

**Review by Nyeriah — APPROVED** [review]

**Comment by Kitzunu** [comment]
I forgot that I only implemented it for characters. I will do it for pets/charters as well

File: src/server/apps/worldserver/worldserver.conf.dist [hunk 2]
```diff
@@ -626,37 +626,37 @@ RealmZone = 1
...
-#                         placed in clientdir/Fonts.
-#                     3 - (Enabled, Basic Latin characters + server timezone specific)
-
-StrictPlayerNames = 0
-
 #
 #   StrictPlayerNames.Reserved
 #       Description: Use the Reserved Filter from DBC.
-#                    Prevents players from creating characters with profanity names.
+#                    Prevents Player, Pet & Charter names from containing reserved names.
```
Comment (heyitsbench) [inline src/server/apps/worldserver/worldserver.conf.dist:632]: Is there much of a reason these are capitalized?

Comment (Kitzunu) [reply src/server/apps/worldserver/worldserver.conf.dist:632]: I see them as names in this context, therefore capitalized

**Comment by Kitzunu** [comment]
Charl creation
image
image

Charter
image

Pet
image
image

**Comment by 55Honey** [comment]
Is there a conf option to read ONLY from the `namesreserved_dbc` and "namesprofanity_dbc` and skip the actual dbc files?

**Pull request merged**